### ISSUE TITLE
`no-extra-parens` has a false negative with binary expressions that are only one level deep

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7229]
**Tell us about your environment**

* **ESLint Version:** 3.6.0
* **Node Version:** 6.6.0
* **npm Version:** 3.10.3

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:**

(none)

**What did you do? Please include the actual source code causing the issue.**

```bash
echo '((((foo)))) instanceof bar' | eslint --stdin --no-eslintrc --rule 'no-extra-parens: [2, "all", {"nestedBinaryExpressions": false}]'
```

**What did you expect to happen?**

I expected ESLint to report an error, because the parentheses around `foo` are unnecessary.

**What actually happened? Please include the actual, raw output from ESLint.**

No error.

**Comment by not-an-aardvark** [comment]
Working on a fix.

**Comment by platinumazure** [comment]
Confirmed the original example in online demo.

@not-an-aardvark Could you be more specific about what "some binary expressions" means here? Thanks!

**Comment by not-an-aardvark** [comment]
Updated the title to be more descriptive. (Additional explanation can also be found in https://github.com/eslint/eslint/pull/7231.)
